evalml/tests/automl_tests/test_automl.py

```diff
@@ -2701,3 +2780,17 @@ def test_automl_issues_beta_warning_for_time_series(problem_type, X_y_binary):
         assert len(warn) == 1
         message = "Time series support in evalml is still in beta, which means we are still actively building its core features"
         assert str(warn[0].message).startswith(message)
+
+
+def test_automl_validates_data_passed_in_to_allowed_pipelines(X_y_binary, dummy_binary_pipeline_class):
+
+    X, y = X_y_binary
+
+    with pytest.raises(ValueError, match="Parameter allowed_pipelines must be either None or a list!"):
+        AutoMLSearch(X, y, problem_type="binary", allowed_pipelines=dummy_binary_pipeline_class)
+
+    with pytest.raises(ValueError, match="Every element of allowed_pipelines must be a subclass of PipelineBase!"):
```

Comment:
This seems like it could be confusing for the user since the pipeline is technically a subclass of `PipelineBase`. I think it'd be much more helpful to output an error stating the class must not be instantiated, since that's the actual issue here

Reply:
Counterpoint: Ut just happened to be that @freddyaboulton  tested using an instantiated object but this error would be raised if we just threw in something random ex: AutoMLSearch class so I think this is fine